docs-src/api-overview.md

```diff
@@ -79,6 +96,132 @@ BackendPolicy. For more information on what may be configured with this resource
 in the future, refer to the related [GitHub
 issue](https://github.com/kubernetes-sigs/service-apis/issues/196).
 
+### Route binding
+
+When a Route binds to a Gateway it represents configuration that is applied on
+the Gateway that configures the underlying load balancer or proxy. How and which
+Routes bind to Gateways is controlled by the resources themselves. Route and
+Gateway resources have built-in controls to permit or constrain how they select
+each other. This is useful for enforcing organizational policies for how Routes
+are exposed and on which Gateways. Consider the following example:
+
+> A Kubernetes cluster admin has deployed a Gateway “shared-gw” in the “Infra”
+> Namespace to be used by different application teams for exposing their
+> applications outside the cluster. Teams A and B (in Namespaces “A” and “B”
+> respectively) bind their Routes to this Gateway. They are unaware of each other
+> and as long as their Route rules do not conflict with each other they can
+> continue operating in isolation. Team C has special networking needs (perhaps
+> performance, security, or criticality) and they need a dedicated Gateway to
+> proxy their application to the outside world. Team C deploys their own Gateway 
+> “dedicated-gw”  in the “C” Namespace that can only be used by apps in the "C" 
+> Namespace.
+
+<!-- source: https://docs.google.com/presentation/d/1neBkFDTZ__vRoDXIWvAcxk2Pb7-evdBT6ykw_frf9QQ/edit?usp=sharing -->
+![route binding](images/gateway-route-binding.png)
+
+There is a lot of flexibility in how Routes can bind to Gateways to achieve
+different organizational policies and scopes of responsibility. These are 
+different relationships that Gateways and Routes can have:
+
+- **One-to-one** - A Gateway and Route may be deployed and used by a single 
+  owner and have a one-to-one relationship. Team C is an example of this. 
+- **One-to-many** - A Gateway can have many Routes bound to it that are owned by
+  different teams from across different Namespaces. Teams A and B are an example 
+  of this.
+- **Many-to-one** - Routes can also be bound to more than one Gateway, allowing 
+  a single Route to control application exposure simultaneously across different
+  IPs, load balancers, or networks.
+
+*In summary, Gateways select Routes and Routes also select Gateways. When both 
```

Comment:
For my own interest, since I am still new to looking at service API. Is right to think that a Gateway defines a behavior it offers and Routes will "choose" which Gateway, with appropriate behavior, the Route logic should apply to? For example, a Gateway that allows public ingress traffic will label itself as public-gw and if you want a network path that can route public traffic to your Service you will label the Route as such so that the Gateway will pick it up?

Reply:
@GnatorX that is correct, a Gateway implements behavior which is defined by a GatewayClass. The characteristics of the GatewayClass define how the Route is exposed (on what infrastructure, where in the network, with what capabilities etc).

@robscott I agree that the way we show Route-to-Gateway selection by using labels that represent one or more Gateways is indirect. However, even though the Gateways define the labels, and Route owners may have no awareness of how these labels map to actual Gateways, the act of putting a label on the Route to influence which Gateways a Route will bind to, is an act of the Route owner selecting the Gateways.

I still think that each persona things of themselves as selecting or limiting Route binding. Gateway owners want to have the power to select or limit which Routes can bind. Route owners also want to have the power to select or limit which Gateways their Route will bind to. I don't think an app owner is comfortable just throwing their Route to the wind and not caring or knowing where their app will be exposed. The app owner wants to influence where and how their app will be exposed on networks, particularly if they have multiple Gateways available for them to choose from. 

Is there any reason why you don't think it makes sense to think of this as both Route->Gateway selection and Gateway->Route selection?

Reply:
@robscott coming back to this, I agree that technically speaking Routes do not select _specific_ Gateways, however I do believe the mental model of Route owners is that they are making an explicit decision that influences which Gateways expose their Routes.

To say that it's the Gateway limiting which Routes select, feels like we are explaining it in a more complicated way than the average user has to know about.

Reply:
Changed this to:

>In summary, Gateways select Routes and Routes allow or deny the selection. When both 
select each other then a Route will bind to a Gateway.